api/v1beta2/ibmvpccluster_types.go

```diff
@@ -103,25 +114,118 @@ type VPCLoadBalancerStatus struct {
 	// +optional
-	Ready       bool        `json:"ready"`
-	Subnet      Subnet      `json:"subnet,omitempty"`
-	VPCEndpoint VPCEndpoint `json:"vpcEndpoint,omitempty"`
+	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
 
 	// ControlPlaneLoadBalancerState is the status of the load balancer.
+	// dep: rely on NetworkStatus instead.
 	// +optional
 	ControlPlaneLoadBalancerState VPCLoadBalancerState `json:"controlPlaneLoadBalancerState,omitempty"`
 
-	// Conditions defines current service state of the load balancer.
+	// COSInstance is the reference to the IBM Cloud COS Instance used for the cluster.
+	COSInstance *ResourceReference `json:"cosInstance,omitempty"`
+
+	// networkStatus is the status of the VPC network in its entirety resources.
+	NetworkStatus *VPCNetworkStatus `json:"networkStatus,omitempty"`
+
+	// ready is true when the provider resource is ready.
+	// +kubebuilder:default=false
+	Ready bool `json:"ready"`
+
+	// resourceGroup is the reference to the IBM Cloud VPC resource group under which the resources will be created.
+	ResourceGroup *ResourceReference `json:"resourceGroupID,omitempty"`
+
+	// dep: rely on NetworkStatus instead.
+	Subnet Subnet `json:"subnet,omitempty"`
+
+	// dep: rely on NetworkStatus instead.
+	VPC VPC `json:"vpc,omitempty"`
+
+	// dep: rely on ControlPlaneEndpoint
+	VPCEndpoint VPCEndpoint `json:"vpcEndpoint,omitempty"`
+}
+
+// VPCNetworkStatus provides details on the status of VPC network resources.
+type VPCNetworkStatus struct {
+	// computeSubnets references the VPC Subnets for the cluster's Data Plane.
 	// +optional
-	Conditions capiv1beta1.Conditions `json:"conditions,omitempty"`
+	ComputeSubnets []*VPCResourceStatus `json:"computeSubnets,omitempty"`
+
+	// controlPlaneSubnets references the VPC Subnets for the cluster's Control Plane.
+	// +optional
+	ControlPlaneSubnets []*VPCResourceStatus `json:"controlPlaneSubnets,omitempty"`
+
+	// loadBalancers references the VPC Load Balancer's for the cluster.
+	// +optional
+	LoadBalancers []VPCLoadBalancerStatus `json:"loadBalancers,omitempty"`
+
+	// publicGateways references the VPC Public Gateways for the cluster.
+	// +optional
+	PublicGateways []*VPCResourceStatus `json:"publicGateways,omitempty"`
+
+	// securityGroups references the VPC Security Groups for the cluster.
+	// +optional
+	SecurityGroups []*VPCResourceStatus `json:"securityGroups,omitempty"`
+
+	// vpc references the IBM Cloud VPC.
+	// +optional
+	VPC *VPCResourceStatus `json:"vpc,omitempty"`
+}
+
+// VPCResourceStatus identifies a resource by crn and type and whether it was created by the controller.
+type VPCResourceStatus struct {
+	// controllerCreated indicates whether the resource is created by the CAPI controller.
+	ControllerCreated bool `json:"controllerCreated,omitempty"`
+
+	// crn defines the IBM Cloud CRN of the resource.
+	// +required
+	CRN string `json:"crn"`
+
+	// type defines the type of IBM Cloud resource.
+	// +required
+	Type ResourceType `json:"type"`
```

Comment:
I think its usage will be like 
````
VPC *VPCResourceStatus `json:"vpc,omitempty"
```

so do you think we need to again set VPC.Type =vpc

Reply:
Yes I think that would be the simple case, I am concerned about more complex cases.

I will drop the type for now then, and attempt to add it back in if/when I need to when I start making changes to `reconcile` logic changes.

Reply:
I dropped the Type for now.